src/radical/pilot/pmgr/launching/base.py

```diff
@@ -595,26 +608,35 @@ def _start_pilot_bulk(self, resource, schema, pilots):
             self._prof.prof('submission_start', uid=pilot['uid'])
 
         # launcher handles pilot job submission
-        launcher = self.find_launcher(rcfg, pilots)
-        launcher.launch_pilots(rcfg, pilots)
+        self.find_launchers(rcfg, pilots)
 
-        for pilot in pilots:
-            self._prof.prof('submission_stop', uid=pilot['uid'])
+        for lname,launcher in self._launchers.items():
+            bucket = list()
+            for pilot in pilots:
+                if pilot['launcher'] == lname:
+                    bucket.append(pilot)
+
+            if bucket:
+                launcher.launch_pilots(rcfg, bucket)
```

Comment:
`rcfg` also not needed here, pilots have all that

Reply:
Hmm, the pilot does not actually have the complete resource config - or am I missing something?

Reply:
it is set within agent's cfg, which then is set to pilot
https://github.com/radical-cybertools/radical.pilot/blob/7f1cfee2b7ebfc2a4b5522159e5739ac18908c85/src/radical/pilot/pmgr/launching/base.py#L880-L884

Reply:
Fair - lets leave it anyway please.  I actually would opt for not adding the agent cfg to the pilot dict.  I think we only did that to get it into MongoDB, but I don't think it has any other purpose really and as such should be removed when we get rid of MongoDB.

Reply:
Also, that contradicts the bulk launching somewhat: we potentially launch several pilots with the same resource config.  If each pilot has it's own we need to assume or ensure that they are the same which seems a bit backwards.

Reply:
ack on leave it for now!

> we potentially launch several pilots with the same resource config.

pilots are split among buckets, thus pilots with the same resource and schema_access are grouped together, that seems legit